### DISCUSSION TITLE
Feature Request: Ability to exclude query param from docs but not openapi

### DISCUSSION THREAD

**Discussion by jordantshaw** [fastapi/fastapi#12267] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query

app = FastAPI()

@app.get("/foo")
def get_foo(query: str = Query(include_in_docs=False):
    return {"foo": "bar"}
```

### Description

I would like to be able to exclude certain query parameters from the docs, without removing it from the openapi spec. Currently we can use `include_in_schema=False` to exclude the parameter from the docs as well as openapi. But I would like to be able to control this speperately. I would suggest something like `include_in_docs`.

### Operating System

macOS

### FastAPI Version

latest

### Pydantic Version

latest

### Python Version

3.12

**Comment by JavierSanchezCastro** [comment]
I don't think it is possible, in any case, you have to check first the openapi documentation to see if it is possible to add an attribute that works as you say

**Comment by jfeaver** [comment]
This is possible but might not be the responsibility of FastAPI. FastAPI generates the OpenAPI JSON and passes it onto Swagger/Redoc. If something is in that JSON then Swagger/Redoc are going to decide when and where to display that information. Perhaps FastAPI could generate two sets of the OpenAPI JSON: One to serve to clients as the raw spec and another to provide to Swagger/Redoc for the UI.

**Comment by YuriiMotov** [comment]
Docs are built from openapi.json. So, if you want to change docs you will have to change openapi schema.

But, you can create two versions of openapi (two endpoints):
* first - for docs only
* second - for external usage
